zipline/finance/slippage.py

```diff
@@ -77,7 +78,45 @@ def fill_price_worse_than_limit_price(fill_price, order):
     return False
 
 
-class SlippageModel(with_metaclass(ABCMeta)):
+class SlippageModelMeta(ABCMeta):
+    """
+    This metaclass allows users to create custom slippage models that support
+    both equities and futures by subclassing EquityFutureModel and
+    FutureSlippageModel together.
+
+    Take for example the following custom model:
+
+        class MyCustomSlippage(EquitySlippageModel, FutureSlippageModel):
+            def process_order(self, data, order):
+                ...
+
+    Ordinarily the first parent class in the MRO ('EquitySlippageModel' in this
+    example) would override the 'allowed_asset_types' class attribute to only
+    include equities. However, this is probably not the expected behavior for a
+    reasonable user, so the metaclass will handle this specific case by
+    manually allowing both equities and futures for the class being created.
+    """
+
+    def __new__(mcls, name, bases, dict_):
+        if 'allowed_asset_types' not in dict_:
+            allowed_asset_types = tuple(
+                chain.from_iterable(
+                    base_class.allowed_asset_types
+                    for base_class in bases
```

Comment:
> I like this! Only comment I have is that in the for loop the word "marker" is slightly confusing in that it's only maybe a marker; might just say "base_class".

Yup, I see your point. The challenge is that there are two uses here: the first is the for-loop var which is an element of `bases`; the second is an instance of `AssetSlippageMarker` which has the `allowed_asset_types` attribute (line above this can make that assumption). I find it surprising to see `base_class.allowed_asset_types`, but you find it confusing to name each element of `bases` as `marker`. Both are valid, so feel free to use what you have here.